### ISSUE TITLE
RFC: rename `isone`, `isequal`, `order` attributes to something better

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/recog#97]
`isone` returns a function that can be used for the purpose one would normally use `IsOne` for. This is a bit confusing... Of course best would be if we did not need these functions at all (which are meant to deal with projective groups, which in `recog` are matrix groups which we implicitly treat as projective, by ignoring scalars.

One idea would be to call them `isonefunc`, `isequalfunc`, `orderfunc`, or perhaps `IsOneFunc`, `IsEqualFunc`, `OrderFunc`. Other ideas?

Taking a step back, of course the best (for the user / method author) solution would be to just remove all of these, and make a new type for "projective matrices", which wraps regular matrices but installs different methods for `IsOne`, `\=` and `Order`. The concern with this that it might cause a significant performance overhead, and also might cause problems in various places (this wrapper would at the very least need to patch through the full `MatrixObj` interface -- which, however, still has not been finalized...)

Another compromise to at least make this a bit nice is to add two argument methods to `IsOne` etc. which take a recognition record as first argument, and a group element as second. Then instead of `ri!.isone(x)` or `isone(ri)(x)`, you might call `IsOne(ri, x)`. Still not as nice as just `IsOne(x)` but at least perhaps slightly less confusing... Or is it?

**Comment by fingolfin** [comment]
Thoughts are welcome (pinging @ssiccha @ChrisJefferson @wilfwilson @DominikBernhardt and everybody else, too :-)

**Comment by ssiccha** [comment]
I would prefer to introduce a new type for "projective matrices". Could the overhead really be that significant? (I have no idea)

If it turns out to be too inefficient I'd go with `IsOneFunction` and syntax `IsOneFunction(ri)(x)`.

In any case getting rid of the `ri!.isone(x)` statements would be really nice.

**Comment by ChrisJefferson** [comment]
I think the best short-term option is the two argument versions.

I've already hit problems where I had code with a variable called 'order', which isn't compatible with the recog package being loaded, so they obviously need changing.

**Comment by fingolfin** [comment]
@ChrisJefferson that, however, is easily fixed, as there isn't really a reason why these things are stored as atttributes.

**Comment by wilfwilson** [comment]
I agree that it would be great to have proper projective objects. But I also agree that that should be a long-term aim.

It's unclear to me whether one of the more short-term suggestions is more/less confusing than the other.

In any case, I think that adding the word "func" or "function"  to the names of these things would be an improvement on the status quo. And in general I would prefer capitalisation rather than the all-lowercase that we have in some places.

**Comment by ssiccha** [comment]
Since these are some semi-hackish attributes I would even call them `RECOG_IsOneFunction` etc.
